103-bundle-runtime.md

```diff
@@ -140,11 +140,12 @@ The parameter value is evaluated thus:
 - If the CNAB runtime does not provide a value, but `default` is set, then the default value MUST be used.
 - If the parameter is marked `required` and `default` is set, then the requirement is satisfied by the runtime-provided default.
 - If no value is provided and `default` is unset, the runtime MUST set the value to an empty string (""), regardless of type.
-- Values are encoded as JSON strings.
+- When serializing non-string environment variable values, types SHOULD be converted to their [JSON equivalent type](https://www.json.org/json-en.html) and expressed in string format. For example, boolean `true` (or `True` or `TRUE`) SHOULD be expressed as `"true"` and the object `{foo: 23}` SHOULD be expressed as `"{\"foo\": 23}"`.
```

Comment:
```suggestion
- Non-string values SHOULD be converted to [JSON text](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf). For example, boolean `true` (or `True` or `TRUE`) SHOULD be expressed as `true` and the object `{foo: 23}` SHOULD be expressed as `{"foo":23}`. String values SHOULD NOT be converted to JSON text.
```

Environment variables and files are both unstructured byte-strings, so serialization is required for both. And I believe that my suggested wording more accurately describes cnab-go's existing behavior.

https://github.com/youreddy/cnab-go/blob/d3645483baf494770ddbc1d83545d20060dd8427/action/action_test.go#L205-L213